packages/core/useCssVar/index.ts

```diff
@@ -27,6 +27,18 @@ export function useCssVar(
   const variable = ref(initialValue)
   const elRef = computed(() => unrefElement(target) || window?.document?.documentElement)
 
+  tryOnMounted(() => {
+    if (elRef.value) {
+      const oldSetProperty = elRef.value.style.setProperty
+      elRef.value.style.setProperty = function (...args) {
+        const [key, value] = args
+        if (key === resolveUnref(prop) && value !== variable.value)
+          variable.value = value!
+        return oldSetProperty.apply(this, args)
+      }
+    }
```

Comment:
This is probably not a good idea to monkey patch the native functions, we either need a way to listen to the change, or we just list as a limitation and discourage the usage of directly setProperty

Reply:
We can't prevent other libraries from using setProperty to modify css var. I saw in the [vue3 source code](https://github.com/vuejs/core/blob/main/packages/runtime-dom/src/helpers/useCssVars.ts) that it uses MutationObserver to monitor changes in css var. Can we also use MutationObserver to monitor?

Reply:
Yeah we should use `MutationObserver`, and adding a flag to enable the watcher so ppl don't need it do pay the cost

Reply:
I added a sync option to enable MutationObserver, which is disabled by default .